packages/settings-library/src/settings_library/email.py

```diff
@@ -18,3 +19,29 @@ class SMTPSettings(BaseCustomSettings):
     SMTP_TLS_ENABLED: bool = Field(False, description="Enables Secure Mode")
     SMTP_USERNAME: Optional[str]
     SMTP_PASSWORD: Optional[SecretStr]
+
+    @root_validator
+    @classmethod
+    def both_credentials_must_be_set(cls, values):
+        username = values.get("SMTP_USERNAME")
+        password = values.get("SMTP_PASSWORD")
+
+        if username is None and password or username and password is None:
```

Comment:
WARNING: make sure the boolean operators priority is right here. I recommend you to use parenthesis:

Can evaluate in a console

```python
import itertools
for username, password in itertools.product([None, "something"]):
     print(username, password, bool(username is None and password or username and password is None) )
```
```

Reply:
ok, i have also seen the test. I guess that is enough to prevent issues if we do changes